src/apps/api/views/search.py

```diff
@@ -35,75 +51,89 @@ def get(self, request, version="v1"):
             "showing_default_results": False,
         }
 
-        if not empty_search:
-            # Setup ES connection, excluding HTML text from our results
-            s = get_search_client()
-
-            # Do search/filtering/sorting
-            s = self._search(s, query)
-            s = self._filter(s, date_flags, start, end, producer)
-            s = self._sort(s, sorting, query)
-            data["results"] = get_results(s)
-
-        if not data["results"] or empty_search:
-            data["showing_default_results"] = True
-            data["results"] = get_default_search_results()
+        print("Empty search?: {}".format(empty_search))
 
+        if not empty_search:
+                object_types = [obj.strip() for obj in object_types.split(',')]
+                s = get_search_client(100, page)
+                s = self._search(s, query, object_types)
+                s = self._filter(s, date_flags, start, end, producer, object_types)
+                s = self._sort(s, sorting, query, object_types)
+                matching_types = []
+                if object_types and object_types != 'ALL' and 'ALL' not in object_types:
+                    for object_type in object_types:
+                        matching_types.append({'match': {'_obj_type': object_type}})
+                else:
+                    for object_type in OBJECT_LIST:
+                        matching_types.append({'match': {'_obj_type': object_type}})
+                s = s.query('bool', should=matching_types, minimum_should_match=1)
+                data["results"] = get_results(s)
         return Response(data)
 
-    def _search(self, search, query):
+    def _search(self, search, query, obj_types='ALL'):
         if query and query != ' ':
+            fields = []
+            if 'competition' in obj_types:
+                fields += ["title^3", "description^2", "html_text^2", "created_by"]
+            if 'user' in obj_types:
+                fields += ["email^3", "username^3", "name^3"]
+            if 'profile' in obj_types:
+                fields += ["email^3", "username^3", "name^3"]
+            if obj_types == 'ALL' or 'ALL' in obj_types:
+                fields += ['title^3', 'username^3', 'name^3']
```

Comment:
I only added common fields between the objects here. I think this makes the most sense since we won't match objects that don't have the field it's trying to match.